diff --git a/sdk/src/OssClientImpl.cc b/sdk/src/OssClientImpl.cc
index 5ba6242..2e9f4d0 100644
--- a/sdk/src/OssClientImpl.cc
+++ b/sdk/src/OssClientImpl.cc
@@ -160,6 +160,10 @@ void OssClientImpl::addBody(const std::shared_ptr<HttpRequest> &httpRequest, con
         httpRequest->setHeader(Http::CONTENT_MD5, md5);
     }

+    if (httpRequest->method() == Http::Method::Put){
+        httpRequest->removeHeader(Http::CONTENT_LENGTH);
+    }
+
     httpRequest->addBody(body);
 }

