diff --git a/bazel/BUILD b/bazel/BUILD
index 3fe42fc7e7..00e91726a6 100644
--- a/bazel/BUILD
+++ b/bazel/BUILD
@@ -20,6 +20,27 @@ licenses(["notice"])

 package(default_visibility = ["//:__subpackages__"])

+config_setting(
+    name = "with_boringssl",
+    values = {"define": "ssl=boringssl"},
+)
+
+alias(
+    name = "ssl",
+    actual = select({
+        ":with_boringssl": "@boringssl//:ssl",
+        "//conditions:default": "@com_github_openssl_openssl//:openssl",
+    }),
+)
+
+alias(
+    name = "crypto",
+    actual = select({
+        ":with_boringssl": "@boringssl//:crypto",
+        "//conditions:default": "@com_github_openssl_openssl//:openssl",
+    }),
+)
+
 filegroup(
     name = "_gevent_test_main",
     srcs = ["_gevent_test_main.py"],
diff --git a/bazel/grpc_deps.bzl b/bazel/grpc_deps.bzl
index 9179107597..fb1db26af1 100644
--- a/bazel/grpc_deps.bzl
+++ b/bazel/grpc_deps.bzl
@@ -72,12 +72,12 @@ def grpc_deps():

     native.bind(
         name = "libssl",
-        actual = "@boringssl//:ssl",
+        actual = "//bazel:ssl",
     )

     native.bind(
         name = "libcrypto",
-        actual = "@boringssl//:crypto",
+        actual = "//bazel:crypto",
     )

     native.bind(
diff --git a/bazel/grpc_extra_deps.bzl b/bazel/grpc_extra_deps.bzl
index c2cf0027a4..9540390900 100644
--- a/bazel/grpc_extra_deps.bzl
+++ b/bazel/grpc_extra_deps.bzl
@@ -51,7 +51,7 @@ def grpc_extra_deps(ignore_version_differences = False):
     api_dependencies()

     go_rules_dependencies()
-    go_register_toolchains()
+    go_register_toolchains(go_version = "host")

     apple_rules_dependencies(ignore_version_differences = ignore_version_differences)
